contracts/facets/ArchaeologistFacet.sol

```diff
@@ -163,58 +155,78 @@ contract ArchaeologistFacet {
     }
 
-    /// @notice Unwraps the sarcophagus.
-    /// @dev Verifies that the unencrypted shard matches the hashedShard stored
-    /// on chain and pays the archaeologist.
+    /// @notice Publishes the raw key share for which the archaeologist is responsible during the
+    /// sarcophagus resurrection window.
+    /// Pays digging fees to the archaeologist and releases their locked bond.
+    /// Cannot be called on a compromised or buried sarcophagus.
     /// @param sarcoId The identifier of the sarcophagus to unwrap
-    /// @param unencryptedShard The archaeologist's corresponding unencrypted shard
-    function unwrapSarcophagus(bytes32 sarcoId, bytes memory unencryptedShard)
-        external
-    {
-        LibUtils.revertIfNotExistOrInactive(sarcoId);
-
-        // Confirm that the archaeologist has not already unwrapped by checking
-        // if the unencryptedShard is empty
-        LibUtils.archaeologistUnwrappedCheck(sarcoId, msg.sender);
-
-        // Confirm that the sender is an archaeologist on this sarcophagus
-        if (!LibUtils.archaeologistExistsOnSarc(sarcoId, msg.sender)) {
-            revert LibErrors.ArchaeologistNotOnSarcophagus(msg.sender);
+    /// @param rawKeyShare The keyshare the archaeologist is publishing
+    function publishKeyShare(bytes32 sarcoId, bytes calldata rawKeyShare) external {
+        LibTypes.Sarcophagus storage sarcophagus = s.sarcophagi[sarcoId];
+
+        // Confirm sarcophagus exists
+        if (sarcophagus.resurrectionTime == 0) {
+            revert LibErrors.SarcophagusDoesNotExist(sarcoId);
+        }
+
+        // Confirm sarcophagus has not been compromised
+        if (sarcophagus.isCompromised) {
+            revert LibErrors.SarcophagusCompromised(sarcoId);
+        }
+
+        // Confirm sarcophagus is not buried
+        if (sarcophagus.resurrectionTime == 2**256 - 1) {
+            revert LibErrors.SarcophagusInactive(sarcoId);
         }
 
-        // Confirm that the resurrection time has passed and that the
-        // grace period has not passed
-        LibUtils.unwrapTime(s.sarcophagi[sarcoId].resurrectionTime);
+        // Confirm current time is after resurrectionTime
+        if (block.timestamp < sarcophagus.resurrectionTime) {
+            revert LibErrors.TooEarlyToUnwrap(sarcophagus.resurrectionTime, block.timestamp);
+        }
+
+        // Confirm current time is within gracePeriod
+        if (block.timestamp > sarcophagus.resurrectionTime + s.gracePeriod) {
+            revert LibErrors.TooLateToUnwrap(
+                sarcophagus.resurrectionTime,
+                s.gracePeriod,
+                block.timestamp
+            );
+        }
+
+        // Confirm tx sender is an archaeologist on the sarcophagus
+        LibTypes.CursedArchaeologist storage cursedArchaeologist = s
+            .sarcophagi[sarcoId]
+            .cursedArchaeologists[msg.sender];
+        if (cursedArchaeologist.doubleHashedKeyShare == 0) {
```

Comment:
Not sure if this is worth having a utility for or just something we need to remember, but checking for the default empty state on `doubleHashedKeyShare` (a `bytes32`) and the default empty state on `rawKeyShare` (a `bytes`) is different in a subtle way:
bytes32 we treat like a number and check for a value of zero
`cursedArchaeologist.doubleHashedKeyShare == 0`
bytes we treat like a string and check for empty length
`cursedArchaeologist.rawKeyShare.length != 0`
_checking for length > 0 on a bytes32 will always return true because the bytes32 is initialized with 32 zeroes ☠️_

Reply:
I will make a utility 😅

Reply:
yep! Something like that